examples/example-react-native-wallet/screens/AuthenticationScreen.tsx

```diff
@@ -0,0 +1,87 @@
+import { PublicKey } from '@solana/web3.js';
+import React from 'react';
+import {NativeModules, Platform, StyleSheet, View} from 'react-native';
+import {Button, Divider, Text} from 'react-native-paper';
+
+import FadeInView from './../components/FadeInView';
+
+const SolanaMobileWalletAdapter =
+    Platform.OS === 'android' && NativeModules.WalletLib
+        ? NativeModules.WalletLib
+        : new Proxy(
+              {},
+              {
+                  get() {
+                      throw new Error(
+                          Platform.OS !== 'android'
+                              ? 'The package `solana-mobile-wallet-adapter-protocol` is only compatible with React Native Android'
+                              : LINKING_ERROR,
+                      );
+                  },
+              },
+          );
+
+type Props = Readonly<{
+  publicKey: PublicKey;
+}>;
+
+export default function AuthenticationScreen({publicKey}: Props) {
+  const [visible, setIsVisible] = React.useState(true);
+
+  // there has got to be a better way to reset the state, 
+  // so it alwyas shows on render. I am react n00b 
+  React.useEffect(() => {
+    setIsVisible(true);
+  });

```

Comment:
By your comment, I'm guessing by "always shows on render", you want the component to be visible whenever the screen is loaded/navigated to?

If I understood correctly, I think a solution is that you can add an empty dependency array to the useEffect to ensure its called only when the component mounts, rather than on every render.

```
React.useEffect(() => {
    setIsVisible(true);
  }, []);
```